source/Applications/Advanced/TransformViaArucoMarker/TransformViaArucoMarker.cpp

```diff
@@ -0,0 +1,321 @@
+        auto normal = u.cross(v);
+        auto unitNormal = normal / cv::norm(normal, cv::NORM_L2);
+
+        auto rotationMatrix = cv::Mat(3, 3, CV_64F);
+        for(int i = 0; i < 3; ++i)
+        {
+            rotationMatrix.at<double>(i, 0) = u[i];
+            rotationMatrix.at<double>(i, 1) = v[i];
+            rotationMatrix.at<double>(i, 2) = unitNormal[i];
+        }
+
+        std::cout << "Displaying ArUco marker's pose" << std::endl;
+        displayMarkerPose(pointCloud, corners2D, center2D);
+
+        return transformationMatrix(rotationMatrix, origin);
+    }
+
+    Zivid::Matrix4x4 invertAffineTransformation(const Zivid::Matrix4x4 &transform)
+    {
+        auto rotation = cv::Mat(3, 3, CV_32F);
+        for(int i = 0; i < 3; ++i)
+        {
+            for(int j = 0; j < 3; ++j)
+            {
+                rotation.at<float>(i, j) = transform(i, j);
+            }
+        }
+
+        auto translation = cv::Mat(3, 1, CV_32F);
+        for(int i = 0; i < 3; ++i)
+        {
+            translation.at<float>(i, 0) = transform(i, 3);
+        }
+
+        cv::Mat newRotation = rotation.inv();
+        cv::Mat newTranslation = -newRotation * translation;
+
+        auto cvInverted = cv::Mat(4, 4, CV_32F, 0.0);
+        for(int i = 0; i < 3; ++i)
+        {
+            for(int j = 0; j < 3; ++j)
+            {
+                cvInverted.at<float>(i, j) = newRotation.at<float>(i, j);
+            }
+        }
+        for(int i = 0; i < 3; ++i)
+        {
+            cvInverted.at<float>(i, 3) = newTranslation.at<float>(i, 0);
+        }
+        cvInverted.at<float>(3, 3) = 1.0;
+
+        auto *ptr = &cvInverted.at<float>(0, 0);
+        auto invertedTransform = Zivid::Matrix4x4(ptr, ptr + 16);
+
+        return invertedTransform;
+    }
+
+    void displayArUcoMarker(cv::Mat bgr, const std::vector<aruconano::Marker> markers)
+    {
+        for(const auto &marker : markers)
+        {
+            marker.draw(bgr);
+        }
+        cv::imshow("Detected ArUco marker", bgr);
+        cv::waitKey(0);
+    }
+
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        const auto dataFile = std::string(ZIVID_SAMPLE_DATA_DIR) + "/ArucoSample.zdf";
+        std::cout << "Reading ZDF frame from file: " << dataFile << std::endl;
+        const Zivid::Frame frame = Zivid::Frame(dataFile);
+        auto pointCloud = frame.pointCloud();
+
+        std::cout << "Converting to OpenCV BGR image format" << std::endl;
+        cv::Mat bgr = pointCloudToCvBGR(pointCloud);
+
+        std::cout << "Detecting ArUco marker in 2D image" << std::endl;
+        auto markers = aruconano::MarkerDetector::detect(bgr);
+
+        std::cout << "Displaying detected ArUco marker" << std::endl;
+        displayArUcoMarker(bgr, markers);
+
+        std::cout << "Estimating detected ArUco marker's pose" << std::endl;
+        const auto transformMarkerToCamera = getAndDisplayMarkerPose(pointCloud, markers[0]);
```

Comment:
In line 303 add something in the comment with regards to 'displaying'.
Maybe clarify in line 300 that it is displayed in 2D image, and here that it is displayed in the point cloud?

Reply:
Both of the displays are in 2D as it is quite difficult to display it in 3D

Display of the detected marker: 
![image](https://user-images.githubusercontent.com/25568033/105040195-d6fa7d80-5a61-11eb-87cf-1675d0a8c084.png)

Display of ArUco pose:
![image](https://user-images.githubusercontent.com/25568033/105040252-e4b00300-5a61-11eb-836b-1f33213ba125.png)

Reply:
Chaged the print statement to 
'std::cout << "Displaying detected ArUco marker in 2D image" << std::endl;' and
'std::cout << "Estimating and displaying detected ArUco marker's pose" << std::endl;' and